platform/viewer/src/components/DicomUpload/DicomUploadProgress.tsx

```diff
@@ -0,0 +1,407 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { ReactElement } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon, ProgressLoadingBar } from '@ohif/ui';
+import DicomFileUploader, {
+  EVENTS,
+  UploadStatus,
+} from '../../utils/DicomFileUploader';
+import { DicomFileUploaderProgressEvent } from '../../utils/DicomFileUploader';
+import DicomUploadProgressItem from './DicomUploadProgressItem';
+import classNames from 'classnames';
+
+export type DicomUploadProgressProps = {
+  dicomFileUploaderArr: DicomFileUploader[];
+  onComplete: () => void;
+};
+
+// The base/initial interval time length used to calculate the
+// rate of the upload and in turn estimate the
+// the amount of time remaining for the upload. This is the length
+// of the very first interval to get a reasonable estimate on screen in
+// a reasonable amount of time. The length of each interval after the first
+// is based on the upload rate calculated. Faster rates use this base interval
+// length. Slower rates below UPLOAD_RATE_THRESHOLD get longer interval times
+// to obtain more accurate upload rates.
+const BASE_INTERVAL_TIME = 15000;
+
+// The upload rate threshold to determine the length of the interval to
+// calculate the upload rate.
+const UPLOAD_RATE_THRESHOLD = 75;
```

Comment:
I guess the whole dicomUpload should move to the cornerstone extension

1. it is using cornersone-wado-image-loader
2. There are configurations that can be done apparently and it makes sense to be inisde extension than core

I'm not sure how do we handle the worklist page though which is in the core. That should go to default extension in my opinion. Same for Local it should be in the default (but not the concern of this PR). It is odd that the viewer layout (grid, panel, toolbar) are all in the extensions and we have worklist in the core

Reply:
I agree and will look into moving the DICOM upload pieces into the cornerstone extension and referencing it via a customization module.

Reply:
Ok the most recent push includes a commit that takes care of this.